challenge-manager/chain-watcher/watcher.go

```diff
@@ -43,9 +43,15 @@ var (
 	assertionConfirmedCounter               = metrics.GetOrRegisterCounter("arb/validator/scanner/assertion_confirmed", nil)
 )
 
+type honestEdge struct {
```

Comment:
Another option would be to add a method to SpecEdge which would wrap itself in a struct that implements VerfiedRoyalEdge. So, instead of having to define it here, we could just call `edge.AsHonest()` instead of `&honestEdge{edge}` below?

Both approaches leave me a little unsatisfied, because it's still up to the `ChallengeTree.AddEdge` method to determine whether or not the edge is honest, so someone *could* accidentally perform either `&honestEdge{edge}` or `edge.AsHonest()` in code scopes where the honesty of the edge hasn't actually been verified yet. I guess my ideal situation is that `ChallengeTree.AddEdge` would actually return a `SpecEdge` that has internal state that marks whether or not it is honest (based on the implementation of `AddEdge`, and, then the caller could do something like:
```
edge, err := chal.honesetEdgeTree.AddEdge(ctx, edge)
if edge.IsHonest() {
    err := w.edgeMaager.TrackEdge(ctx, edge.AsVerifiedRoyalEdge())
    ...
```

Reply:
The original idea was to lock it down such that it would not be possible to construct a VerifiedHonestEdge without a function that would hand it to you, but we can't really prevent people from implementing the interface externally to satisfy a function if truly desired. I do like the idea of a method to convert, and having the internal state of the struct tell us if it is honest. We can't prevent others from turning an edge into verified royal by simply creating a wrapper struct that implements the VerifiedRoyalEdge interface, however.